chain/client/src/sync/state.rs

```diff
@@ -535,61 +520,33 @@ impl StateSync {
     /// Only select candidates that we have no pending request currently ongoing.
     fn possible_targets(
         &mut self,
-        me: &Option<AccountId>,
         shard_id: ShardId,
-        chain: &Chain,
-        epoch_manager: &dyn EpochManagerAdapter,
-        sync_hash: CryptoHash,
         highest_height_peers: &[HighestHeightPeerInfo],
-    ) -> Result<Vec<AccountOrPeerIdOrHash>, near_chain::Error> {
-        let prev_block_hash = *chain.get_block_header(&sync_hash)?.prev_hash();
-        let epoch_hash = epoch_manager.get_epoch_id_from_prev_block(&prev_block_hash)?;
-
-        let block_producers =
-            epoch_manager.get_epoch_block_producers_ordered(&epoch_hash, &sync_hash)?;
-        let peers = block_producers
+    ) -> Result<Vec<PeerId>, near_chain::Error> {
+        let peers = highest_height_peers
             .iter()
-            .filter_map(|(validator_stake, _slashed)| {
-                let account_id = validator_stake.account_id();
-                if epoch_manager
-                    .cares_about_shard_from_prev_block(&prev_block_hash, account_id, shard_id)
-                    .unwrap_or(false)
-                {
-                    // If we are one of the validators (me is not None) - then make sure that we don't try to send request to ourselves.
```

Comment:
Does this still work after the refactor?

Reply:
Yes, because the list of peers excludes 'ourselves', and we no longer refer to potential targets by account id.